### PR TITLE
Added OCR typo transformation

### PR DESCRIPTION
**Pull request by goyalpramod**
## Description

Added OCR typo transformation. I have only added replacement and deletion as they seem to be the most common type of error in OCR as mentioned [here](https://aclanthology.org/W96-0108.pdf)

## Related Issue

#1565 

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [x] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: giskard/scanner/robustness/text_transformations.py [hunk 1]
```diff
@@ -122,7 +122,49 @@ def _random_key_typo(self, char):
...
+        from .entity_swap import ocr_typos
+
+        self.rate = rate
+        self.min_length = min_length
+        self._ocr_typos = ocr_typos
+        self.rng = np.random.default_rng(seed=rng_seed)
+
+    def make_perturbation(self, x):
+        # Skip if the text is too short
+        if len(x) < self.min_length:
```
Comment (andreybavt) [inline giskard/scanner/robustness/text_transformations.py:139]: there should also be a `None` check, otherwise it'll fail at `len(None)`

**Review by andreybavt — CHANGES_REQUESTED** [review]

File: tests/scan/test_text_transformations.py [hunk 2]
```diff
@@ -247,3 +247,11 @@ def test_typo_transformation():
...
 
     assert p == "If one doesn't know his misakes, he won't want to corrcet them."
+
+def test_ocr_typo_transformation():
+    from giskard.scanner.robustness.text_transformations import TextFromOCRTypoTransformation
+
+    t = TextFromOCRTypoTransformation(column="text", rng_seed=1)
+    p = t.make_perturbation("If one doesn't know his mistakes, he won't want to correct them.")
+
+    assert p == "1f 0ne d0ezn't kn0uu his nnistakes, he w0n't want to correct tbenn."
```
Comment (andreybavt) [inline tests/scan/test_text_transformations.py:257]: The test doesn't pass for me:

```
Expected :"1f 0ne d0ezn't kn0uu his nnistakes, he w0n't want to correct tbenn."
Actual   :"If one doesn't know his mi5takes, he won't want to corrct them."

```

Comment (andreybavt) [reply tests/scan/test_text_transformations.py:257]: Could you also add a test for the `len(x) < self.min_length` condition?

**Review by andreybavt — DISMISSED** [review]

File: tests/scan/test_text_transformations.py (hunk 2 above)
Comment (goyalpramod) [reply tests/scan/test_text_transformations.py:257]: I will check it and get it done.

**Comment by goyalpramod** [comment]
@andreybavt I apologize for the late fix, I had gone on a new year vacation. Happy New Year. I have fixed the error (I was using another random seed in testing). And added the things you recommend. Kindly let me know if I should do anything else, Thank you.

**Review by kevinmessiaen — APPROVED** [review]

**Comment by kevinmessiaen** [comment]
Thank

> @andreybavt I apologize for the late fix, I had gone on a new year vacation. Happy New Year. I have fixed the error (I was using another random seed in testing). And added the things you recommend. Kindly let me know if I should do anything else, Thank you.

Thanks @goyalpramod and happy new year! No worries about it, hope enjoyed your vacation!

**Pull request merged**